### PR TITLE
Compile `timescaledb`

### PR DESCRIPTION
**Pull request by vadim2404**
## Describe your changes

```
postgres=#
CREATE TABLE stocks_real_time (
  time TIMESTAMPTZ NOT NULL,
  symbol TEXT NOT NULL,
  price DOUBLE PRECISION NULL,
  day_volume INT NULL
);
CREATE TABLE
postgres=# SELECT create_hypertable('stocks_real_time','time');
       create_hypertable
-------------------------------
 (2,public,stocks_real_time,t)
(1 row)

postgres=#
CREATE INDEX ix_symbol_time ON stocks_real_time (symbol, time DESC);
CREATE INDEX
```

After @MMeent's PG upgrade, I was able to compile it for PG14

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [x] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

### CODE REVIEW

File: control_plane/src/compute.rs [hunk 1]
```diff
@@ -319,7 +319,7 @@ impl PostgresNode {
             // uses only needed variables namely host, port, user, password.
             format!("postgresql://no_user:{password}@{host}:{port}")
         };
-        conf.append("shared_preload_libraries", "neon");
+        conf.append("shared_preload_libraries", "neon, timescaledb");
```
[the remark this answers is not in the document]
Comment (vadim2404) [reply control_plane/src/compute.rs:322]: I wasn't able to create extension without this option

[the remark this answers is not in the document]
Comment (kelvich) [reply control_plane/src/compute.rs:322]: That would load timescaledb always. Basically we have two options here: 
* add a possibility to tick extensions that require `shared_preload_libraries` in the console before starting the compute (and restart it if you want to enable extension)
* always preload all libraries

second is easier and no need in restart, but a bit more memory-hungry.

But in any case no need to add it in `control_plane/src/compute.rs`, rather here: https://github.com/neondatabase/cloud/blob/main/goapp/internal/postgressettings/settings.go#L87

**Comment by vadim2404** [comment]
> Specifically, I think we should set at least [`timescaledb.telemetry_level = off`](https://docs.timescale.com/timescaledb/latest/how-to-guides/configuration/timescaledb-config/#timescaledbtelemetry_level-enum)

tried to disable telemetry while compiling it

@hlinnaka 
We discussed with you about using features only under the apache-2 license.
After compilation, I saw these in the logs

```
2023-03-02 10:05:04.427 UTC [883] ERROR:  functionality not supported under the current "apache" license
2023-03-02 10:05:04.427 UTC [883] HINT:  Upgrade your license to 'timescale' to use this free community feature.
```

File: control_plane/src/compute.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vadim2404) [reply control_plane/src/compute.rs]: I've created a PR in the cloud repo.

neondatabase/cloud#4126

**Comment by vadim2404** [comment]
> > 2023-03-02 10:05:04.427 UTC [883] ERROR:  functionality not supported under the current "apache" license
> > 2023-03-02 10:05:04.427 UTC [883] HINT:  Upgrade your license to 'timescale' to use this free community feature.
> 
> Some features of Timescale are not supported under the apache license, so if you have used those features in a cluster before configuring `-DAPACHE_ONLY=ON`, they will now throw errors.

I understood this. It was just note :)

**Pull request merged**